src/components/datepicker/DatePickerInput.tsx

```diff
@@ -138,9 +138,17 @@ function DatePickerInput<IsTypeable extends boolean = true>({
     [inputValue],
   );
   useEffect(() => {
-    setSelectedDate(others.selectedDate);
-    handleInputValueChange(others.selectedDate ? getFormattedDateString(others.selectedDate) : '');
-  }, [others.selectedDate]);
+    if (!preselectedDate) {
+      handleInputValueChange('');
+      setSelectedDate(preselectedDate);
+      // If preseleced invalid date happens when typed date is invalid and value is passed from form in codebase
```

Comment:
I wanted to correct the typo `preseleced` but I didn't manage to understand what this comment is meant to explain 😕

Reply:
Wow that was huge pile of 💩 , fixed